Refactor DBus Connection Initialization

purejava opened the pull request:
Fixes cryptomator#6

overheadhunter commented on src/main/java/org/cryptomator/linux/keychain/KDEWalletKeychainAccess.java:
```diff
@@ -80,6 +63,28 @@ public ConnectedWallet(DBusConnection connection) {
+			try {
+				return DBusConnection.getConnection(DBusConnection.DBusBusType.SESSION);
+			} catch (RuntimeException e) {
+				if ("Cannot Resolve Session Bus Address".equals(e.getMessage())) {
```
This is very prone to unnoticed changes in upstream exception handling. If the message changes, the code stops working.

Can we instead get add an explicit (checked) exception to DBusConnection?

infeo replied:
This is not possible, see https://github.com/hypfvieh/dbus-java/issues/128.

This issue is already handled in dbus-java (without a new release), but the changed also needs to find its way into kdewallet.

overheadhunter replied:
Ok. Would you say we can wait with this PR until we get an upstream version without the RuntimeException?

overheadhunter reviewed (changes requested):
oh wait, didn't see the pom.xml conflict :D please do a reverse merge first

purejava commented:
I think we need a new version of secret-service first, don't we? 😃

overheadhunter commented:
> I think we need a new version of secret-service first, don't we? 😃

Do we? The suggested change affects KDE only, so...?

purejava commented:
> Do we? The suggested change affects KDE only, so...?

Well, Infeo [asked](https://github.com/cryptomator/integrations-linux/issues/5#issuecomment-819364393) both of us for an update.

And as far as I know you @overheadhunter, having secret-service and kdewallet depend on two different versions of dbus-java would lead to two dbus-java jar files within the AppImage and therefore unneccessarily wasted disc space. :grin::stuck_out_tongue: (sorry, could not hold back that one)

I wonder if Maven can manage that, but I think so.

overheadhunter commented:
Well, Maven will pull in just one version. So as long as the API didn't change, we should be fine.
I'd prefer one PR per implementation. But we better merge both at the same time, just in case.

swiesend commented:
I released [1.6.1](https://github.com/swiesend/secret-service/tree/v1.6.1), which depends on `dbus-java` `3.3.0` and fixes the auto-close behavior for https://github.com/swiesend/secret-service/issues/26, but that should not affect the usage in any way.

@purejava can you also remove the `ExceptionInInitializerError | RuntimeException` here please: https://github.com/cryptomator/integrations-linux/blob/7c34294cca919165447c9f63a421c989baf08f66/src/main/java/org/cryptomator/linux/keychain/SecretServiceKeychainAccess.java#L27

and I think you have to rebase, too: `This branch is 4 commits ahead, 19 commits behind cryptomator:develop.`

purejava commented:
That should be it.
I did a `git pull --rebase cryptomator/integrations-linux develop`, changed the catch clause as @swiesend requested and adjusted the dependency versions in the pom. I am pretty confident, that I've chosen the right ones :sweat_smile:.

On solving the merge conflict in the pom I did a mistake and reverted the removed Bintray entries in the pom. That's the reason for the last commit.

swiesend commented:
>  I am pretty confident, that I've chosen the right ones 😅.

Because you have seen what I did in [`1.6.0`](https://github.com/swiesend/secret-service/blob/master/CHANGELOG.md) and nobody would be able to use it?! 😅 I even introduced a regression test to not be able to remove the `public` keyword from the `SimpleCollection` constructor again by accident...

@overheadhunter sorry for the off-topic: Please, let me know if there are any problems with `1.6.1`, as I am currently not able to build a Cryptomator version from `master` with java `16`, that would start properly. The log is fine, but the _JavaFX Application_ does **not** show up at all. Is this a know issue as I could not find an open one?! The only sensible hint I have is, that my system may need additional GTK libs installed: `Gtk-Message: 16:01:43.900: Failed to load module "canberra-gtk-module"`, but I did that and it still does not work.

overheadhunter commented:
> I am currently not able to build a Cryptomator version from master with java 16, that would start properly. The log is fine, but the JavaFX Application does not show up at all. Is this a know issue as I could not find an open one?! The only sensible hint I have is, that my system may need additional GTK libs installed: Gtk-Message: 16:01:43.900: Failed to load module "canberra-gtk-module", but I did that and it still does not work.

Nothing that I am aware of. Does the latest AppImage run? If yes, you should be able to `--appimage-extract` it and simply replace the required jars. Thanks to the service loader api it might even be hot-swappable 😄 (not sure if Cryptomator initializes singleton instances of the keychain stuff, though)

purejava commented:
It's building fine on Arch linux:
```
ralph@fusion main % java -version                    
java version "16" 2021-03-16
Java(TM) SE Runtime Environment (build 16+36-2231)
Java HotSpot(TM) 64-Bit Server VM (build 16+36-2231, mixed mode, sharing)
ralph@fusion main % pwd                              
/home/ralph/git/cryptomator/main
ralph@fusion main % mvn clean install -Plinux,release 
...
```
```
[INFO] --- maven-assembly-plugin:3.3.0:single (assemble-linux) @ buildkit ---
[INFO] Reading assembly descriptor: assembly-linux.xml
[INFO] Building zip: /home/ralph/git/cryptomator/main/buildkit/target/buildkit-linux.zip
[INFO] 
[INFO] --- maven-install-plugin:2.4:install (default-install) @ buildkit ---
[INFO] Installing /home/ralph/git/cryptomator/main/buildkit/pom.xml to /home/ralph/.m2/repository/org/cryptomator/buildkit/1.5.14/buildkit-1.5.14.pom
[INFO] Installing /home/ralph/git/cryptomator/main/buildkit/target/buildkit-linux.zip to /home/ralph/.m2/repository/org/cryptomator/buildkit/1.5.14/buildkit-1.5.14.zip
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Cryptomator 1.5.14:
[INFO] 
[INFO] Cryptomator ........................................ SUCCESS [  0.252 s]
[INFO] Cryptomator Commons ................................ SUCCESS [  6.312 s]
[INFO] Cryptomator GUI .................................... SUCCESS [  7.346 s]
[INFO] Cryptomator Launcher ............................... SUCCESS [  3.688 s]
[INFO] Cryptomator Build Kit .............................. SUCCESS [  1.084 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18.819 s
[INFO] Finished at: 2021-04-18T12:32:53+02:00
[INFO] ------------------------------------------------------------------------
ralph@fusion main % 
```
What is your system @swiesend?

infeo commented:
> I'd prefer one PR per implementation. But we better merge both at the same time, just in case.

I second that. Just found another issue in secret-service 1.6.1. @purejava can you revert bumping secret-service dependency?

purejava commented:
> I second that. Just found an issue in secret-service 1.6.1. @purejava can you revert bumping secret-service dependency?

Sure. Which one is the right one? Commit 3d5ab972d6cd89982f4a5606fc6bb9f78dd4d65f changed the version from **1.5.0** to **1.6.1**. On GitHub, these were the latest versions. On maven central, there are **1.5.0, 1.6.0** and **1.6.1**.

What's the best one to choose @swiesend?
Thanks.

swiesend commented:
If there are problem with 1.6.1 we have to go back to 1.5.0. 1.6.0 was an accident.

> Just found an issue in secret-service 1.6.1
@infeo what is the issue?

swiesend commented:
`secret-service` [`1.6.2`](https://github.com/swiesend/secret-service/blob/v1.6.2/CHANGELOG.md) is released.

purejava commented:
Hm, after reverting the secret-service dependency to 1.5.0, it cannot find the `DBusConnectionException` symbol any more, which comes with dbus-java 3.3.0. I don't know, why this happens.
